urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/ModuleCommandFactory.java

```diff
@@ -0,0 +1,92 @@
+package ikrs.httpd;
+
+import ikrs.typesystem.BasicType;
+import ikrs.util.AbstractCommand;
+import ikrs.util.DefaultCommandFactory;
+import ikrs.util.Command;
+import ikrs.util.UnknownCommandException;
+import ikrs.util.CommandStringIncompleteException;
+
+
+/**
+ * Problem: the ModuleCommand/~Factory would require an HTTPHandler instance to be passed on
+ *          system start. But the system instantiates the HTTPHandler at a later point by
+ *          added a new handler to the yucca server.
+ *          So the factory requires the handler on instantiation and vice versa.
+ *
+ * To solve this problem the handler stores the first instance that was created in a static 
+ * attribute. When called the ModuleCommand tries to access this field.
+ *
+ * Warning: this only works if HTTP config uses a shared instance for _all_ listening ports.
+ *          See the 'sharedHandlerInstance' attribute in the ikrs.httpd.conf file (server
+ *          node).
+ *          If the 'sharedHandlerInstance' is not set Yucca will create different handler
+ *          instances for each 'server' tag. In this case the command will only work
+ *          for the first handler instance created.
+ *
+ * @author Ikaros Kappler
+ * @date 2013-01-09
+ * @modified 2013-04-17 Ikaros Kappler (shared handler instance added).
+ * @version 1.0.0
+ **/
+
+
+public class ModuleCommandFactory
+    extends DefaultCommandFactory {
+
+    public ModuleCommandFactory() {
+	
+	super();
+
+	this.addSupportedCommand( new ModuleCommand( "HTTPD",
+						     null,  // params
+						     0      // offset
+						     )
+				  );
+						   
+
+    }
+
+    //protected HTTPHandler getHandler() {
+    //	return this.handler;
+    //}
+
+    //--- BEGIN --------------------- AbstractCommandFactory implementation ------------------
+    /**
+     * Make a new Command with the given name and params.
+     *
+     * @param name The command's name.
+     * @param params The command's params - in BasicType representation.
+     * 
+     * @return The new command.
+     **/
+    public ModuleCommand make( String name,
+			       BasicType[] params )
+	throws UnknownCommandException,
+	       CommandStringIncompleteException {
+
+	
+	//return null;
+	if( name == null )
+	    throw new NullPointerException( "Command must not be null." );
+	
+	if( name.equalsIgnoreCase("HTTP") || name.equalsIgnoreCase("HTTPD") ) {
+
+	    if( params.length == 0 )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;params&#39; (index: 1 | type: BasicType[). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;name&#39;. The same parameter type is validated here for example: urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/ModuleCommand.java:35. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).